sotodlib/tod_ops/fft_ops.py

```diff
@@ -356,6 +359,18 @@ def calc_wn(aman, pxx=None, freqs=None, low_f=5, high_f=10):
     if pxx is None:
         pxx = aman.Pxx
 
+    # limit upper frequency cutoffs to hwp freq
+    if 'hwp_angle' in aman._fields:
+        hwp_freq = (np.sum(np.abs(np.diff(np.unwrap(aman.hwp_angle)))) /
+                     (aman.timestamps[-1] - aman.timestamps[0])) / (2 * np.pi)
+        hwp_freq_limit = 0.95 * hwp_freq
+        if high_f >= hwp_freq and hwp_freq > 0:
+            logger.warning(f"high frequency cutoff={high_f} > hwp_freq={hwp_freq}. Limiting to hwp_freq.")
+            high_f = hwp_freq_limit
+            # make sure low_f < high_f
+            if low_f >= hwp_freq_limit:
```

Comment:
low_f > high_f is a check that should happen regardless of the check on hwp spin rate so I think this check/error raising should happen outside of the hwp_angle field check block.